### ISSUE TITLE
Feature request: Add transfer_queue into VulkanoContext in vulkano-util

### ISSUE THREAD

**Issue by qianzhang5** [vulkano-rs/vulkano#2565]
Adding a transfer_queue into VulkanoContext, which can be used to achieve better performance in upload image/texture data , or other exchanging data cases in a separated thread.

**Comment by John-Nagle** [comment]
I'm using WGPU, and need a transfer queue over there. I have concurrent asset loading from other threads, and it hits the render thread hard and kills the frame rate when assets are being loaded. 
If you do this in Vulkano, maybe WGPU will pick up the code. Or I might switch to Vulkano.

**Comment by EmileGr** [comment]
@John-Nagle It's already possible to use a dedicated transfer queue in Vulkano. Take a look at the async-update example in the examples directory. It's a simple example of exactly what you're describing. This feature request is specifically for adding such a queue to VulkanoContext, which just gives you access to queues, the memory allocator and the device with a little less boilerplate.

**Comment by Rua** [comment]
I'm implementing this, but I wonder what should be done if there is not dedicated transfer queue available on the device. Should the function return `None`, or should it return the graphics/compute queue. In the latter case, what if the graphics and compute queues are different? Is there a situation where you'd specifically want to distinguish a "graphics transfer queue" and a "compute transfer queue"?

**Comment by theprotonfactor** [comment]
In the event that there isn't a dedicated transfer queue then you could perhaps have the transfer_queue function take a QueueFlags as input for what it should return as a fallback? If the user passes in QueueFlags::TRANSFER then you return None otherwise you return whatever they asked for as a fallback.

**Comment by Rua** [comment]
Then you might as well return `Option<&Arc<Queue>>` and let the user do `c.transfer_queue().or(c.graphics_queue())` themselves.

**Comment by theprotonfactor** [comment]
Yeah, that's a little more clean, for sure. I think I do like that because it both signals that there isn't a dedicated transfer queue available and forces the user to decide for themselves what they'd like as an alternative. Though most of time you'd probably be happy with a fallback to the graphics queue, so I could totally see someone else having a different perspective on that.

**Comment by John-Nagle** [comment]
> Then you might as well return `Option<&Arc<Queue>>` and let the user do `c.transfer_queue().or(c.graphics_queue())` themselves.

Agreed. The application should know if there's a transfer queue. If you want a transfer queue, it's because you want to overlap asset loading with rendering. You probably have an asset fetcher pushing assets across the transfer queue as fast as it will go. If the device only has one queue, which is true of some low-end devices, maximizing transfers will slow rendering. The application is going to have to throttle. So it needs to know.
